### PR TITLE
feat: :rocket: Created GitHub OIDC role module

### PR DESCRIPTION
**Pull request by Tanveer143s**
## what
* Created github oidc role terraform module

### CODE REVIEW

**Comment by nileshgadgi** [comment]
Update the example folder name with the official structure.
https://developer.hashicorp.com/terraform/language/modules/develop/structure
from _example to examples

**Comment by Tanveer143s** [comment]
> update this PR branch with base branch (master) pull from master branch

Done

**Comment by Tanveer143s** [comment]
> Update the example folder name with the official structure. https://developer.hashicorp.com/terraform/language/modules/develop/structure from _example to examples

Done

**Review by nileshgadgi — DISMISSED** [review]
Looks Good To Me!

File: .github/workflows/tf-checks.yml [hunk 1]
```diff
@@ -8,4 +8,9 @@ jobs:
   complete-example:
     uses: clouddrove/github-shared-workflows/.github/workflows/tf-checks.yml@1.2.2
     with:
-      working_directory: './_example/complete/'
+      working_directory: './examples/complete/'
+  oidc-example:
+    uses: clouddrove/github-shared-workflows/.github/workflows/tf-checks.yml@1.2.1
```
Comment (VishwajitNagulkar) [inline .github/workflows/tf-checks.yml:13]: use latest tag for tfcheck

Comment (Tanveer143s) [reply .github/workflows/tf-checks.yml]: done

**Review by nileshgadgi — APPROVED** [review]

File: modules/aws_github_oidc_role/variables.tf [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+variable "provider_url" {
+  type        = string
+  description = "The URL of the identity provider for the OIDC"
+}
+
+variable "custom_assume_role_policy" {
+  type        = string
+  default     = ""
+  description = "The URL of the identity provider for the OIDC"
```
Comment (VishwajitNagulkar) [inline modules/aws_github_oidc_role/variables.tf:43]: Plz update the description

**Review by h1manshu98 — APPROVED** [review]
Reviewed Custom `assume_role_policy`, its working as expected.

File: modules/aws_github_oidc_role/variables.tf (hunk 2 above)
Comment (Tanveer143s) [reply modules/aws_github_oidc_role/variables.tf]: done

File: examples/github-oidc/main.tf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+##----------------------------------------------------------------------------- 
+## GitHub OIDC role module call.   
+##-----------------------------------------------------------------------------
+
+module "aws_github_oidc_role" {
+  source = "../../modules/aws_github_oidc_role"
+
+  enable               = true
```
Comment (VishwajitNagulkar) [inline examples/github-oidc/main.tf:17]: why `enable` attribute here because you are not calling github oidc module in main IAM role module

Comment (Tanveer143s) [reply examples/github-oidc/main.tf]: fixed

File: modules/aws_github_oidc_role/main.tf [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  url             = var.provider_url
+  tags            = module.labels.tags
+}
+
+##-----------------------------------------------------------------------------
+## Include the role resource and attachment here
+##-----------------------------------------------------------------------------
+
+resource "aws_iam_role" "github" {
+  count = var.enable ? 1 : 0
```
Comment (VishwajitNagulkar) [inline modules/aws_github_oidc_role/main.tf:52]: why `enable`  attribute we required?

File: examples/github-oidc/main.tf [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+##----------------------------------------------------------------------------- 
+## GitHub OIDC role module call.   
+##-----------------------------------------------------------------------------
+
+module "aws_github_oidc_role" {
+  source = "../../modules/aws_github_oidc_role"
+
+  enable               = false
```
Comment (VishwajitNagulkar) [inline examples/github-oidc/main.tf:17]: why `false` ?

File: modules/aws_github_oidc_role/main.tf (hunk 4 above)
Comment (Tanveer143s) [reply modules/aws_github_oidc_role/main.tf:52]: getting errors when removed

File: examples/github-oidc/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+output "arn" {
+  value       = module.aws_github_oidc_role
+  description = "The Amazon Resource Name (ARN) specifying the role."
+}
+
+output "tags" {
+  value       = module.aws_github_oidc_role
+  description = "A mapping of tags to assign to the resource."
+}
```
Comment (VishwajitNagulkar) [inline examples/github-oidc/outputs.tf:14]: Update output files with proper values

**Comment by nileshgadgi** [comment]
- How is this readme file going to be dynamically updated like the root readme? did you prepared any workflow for this?

**Comment by Tanveer143s** [comment]
> * How is this readme file going to be dynamically updated like the root readme? did you prepared any workflow for this?

no

**Comment by nileshgadgi** [comment]
Then it's not going to be updated dynamically, you can update readme workflow for module readme. take reference from addon module.

**Review by h1manshu98 — APPROVED** [review]

**Comment by Tanveer143s** [comment]
> Then it's not going to be updated dynamically, you can update readme workflow for module readme. take reference from addon module.

we need this module for blog can we do it later in another branch

File: modules/aws_github_oidc_role/README.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
+# Terraform Configuration for AWS IAM GitHub OIDC Role
```
[the remark this answers is not in the document]
Comment (nileshgadgi) [reply modules/aws_github_oidc_role/README.md:1]: he will do it in another branch since it's required for blog publish

**Review by anmolnagpal — APPROVED** [review]

**Pull request merged**